docs: add JavaScript minify & transform example and api docs (#10206)
web-infra-dev/rspack: website/docs/en/api/javascript-api/swc.mdx

# SWC API

> Rspack supports JavaScript minify and transform.
> 
> Rspack uses SWC under the hood to transform and minify JavaScript code, you can access these functions using the `rspack.experiments.swc` API.

Reviewer: ```suggestion
Rspack uses [SWC](https://swc.rs/) under the hood to transform and minify JavaScript code, and experimentally exposes some SWC JavaScript APIs through `rspack.experiments.swc`. This allows you to directly call SWC functions like `transform` or `minify` without installing the additional [@swc/core](https://www.npmjs.com/package/@swc/core) package.
```

## Examples

Here is a simple example demonstrating how to transform JavaScript code using Rspack:

```js
import { rspack } from '@rspack/core';

> const swc = rspack.experiments.swc;

Reviewer: ```suggestion
const { swc } = rspack.experiments;
```

>   .transform('const a:number = 10;', {

Reviewer: ```suggestion
  .transform('const a: number = 10;', {
```

swc
  .transform('const a:number = 10;', {
    filename: 'main.ts',
    jsc: {
      parser: {
        syntax: 'typescript',
      },
      target: 'es5',
    },
  })
  .then(result => {
    console.log(result.code);
  });
```

In addition to using the `swc` API directly, you can also use it in loaders or plugins to help with code transform or minify.

```js
import { rspack } from '@rspack/core';

> const swc = rspack.experiments.swc;

Reviewer: ```suggestion

const { swc } = rspack.experiments;
```

> ```js

Reviewer: ```suggestion
```js title="my-loader.mjs"
```
